gremlin-core/src/main/java/org/apache/tinkerpop/gremlin/process/traversal/step/map/GroupCountStep.java

```diff
@@ -41,7 +41,7 @@
 /**
  * @author Marko A. Rodriguez (http://markorodriguez.com)
  */
-public final class GroupCountStep<S, E> extends ReducingBarrierStep<S, Map<E, Long>> implements TraversalParent, ByModulating {
+public class GroupCountStep<S, E> extends ReducingBarrierStep<S, Map<E, Long>> implements TraversalParent, ByModulating {
```

Comment:
We had originally made many of the steps `final` in the early days of TinkerPop 3.x to prevent extension by providers as they were changing frequently and we wanted to be able to introduce changes without worrying about breaking APIs. I'd imagine that most of the steps have largely stabilized at this point so perhaps that worry is no longer as much of a concern.  However, I still feel like these core step classes are TinkerPop's implementations and I'd not want to see our development blocked for fear of a breaking API change that affects providers (a behavioral change is different as it affects users and we would obviously be sensitive to that). 

Perhaps it would be good to start a DISCUSS thread on this on the dev list to see if there is any concerns with removing `final` for all steps with the idea that breaking API changes become a possibility for providers on patch releases if they choose to directly extend these core step classes. wdyt?

Reply:
Let me change the scope of this PR to only change visibility of certain members. We can pick up removing the final keyword after a discussion in email list.

Reply:
Resolving this comment since I removed the file being discussed from the scope of this PR.